bokeh/models/widgets/buttons.py

```diff
@@ -133,12 +135,13 @@ def on_click(self, handler: Callable[[bool], None]) -> None:
 
         Returns:
             None
-
         """
+        deprecated((3, 0, 0), 'on_click(handler)', 'on_event("button_click", handler)')
         self.on_change('active', lambda attr, old, new: handler(new))
```

Comment:
This (and elsewhere) should change to use the event, to be consistent with other widgets that emit button click events I think

Reply:
Wouldn't that change the behavior of a method that we are intending to be deprecated?